apps/test-app/frontend/src/components/tree-widget/stateless/Tree.tsx

```diff
@@ -161,6 +181,32 @@ export function StatelessTreeWidget(props: Omit<TreeWidgetProps, "rulesetId">) {
   );
 }
 
+interface FormatterSetterDropdownProps {
+  shouldUseCustomFormatter: boolean;
+  setShouldUseCustomFormatter: (value: React.SetStateAction<boolean>) => void;
```

Comment:
Nitpick: I would change `setShouldUseCustomFormatter: (value: React.SetStateAction<boolean>) => void;` to `toggleCustomFormatter: () => void` to avoid directly changing parent component state in child component.

Reply:
Changed